### ISSUE TITLE
BUGs while signing up with Facebook

### ISSUE THREAD

**Issue by adrienjoly** [openwhyd/openwhyd#97]
Reported by [Benjamin Alijagić](https://spectrum.chat/users/benjam1n) on Spectrum (DM):

> hey Adrien whatsup with FB signup on openwhyd
> is it supposed to only prefil my fullname?
> I was expecting to be fully registed after clicking it

![image](https://user-images.githubusercontent.com/531781/30413675-c1bca72e-991f-11e7-8773-180e9ca544bd.png)

> also :D
> might wanna use something like `/\A[^@\s]+@[^@\s]+\z/` for validation

**Comment by znk** [comment]
Provided Regex only applies to Ruby. It also seems Regex mail validation is not as easy as it looks like...
It would be better in future to delegate validation job to third party library?

[From](https://github.com/openwhyd/openwhyd/blob/d5935d7f1001e682174fdcba6b79cbda23e97ecc/whydJS/app/models/email.js#L15)
```js
var emailCheck = /^[\w\.\+\-]+@[a-z0-9\.\-]+\.[a-z]{2,4}$/i;

validate = function(email) {
	if (!email) return email;
	if (typeof email != "string") {
		console.error("(malicious?) non-string email:", email, new Error().stack);
		return "";
	}
	return emailCheck.test(email);
}

validate("mail@domain.com")
true
validate("tools@symbol.agency")
false
```

[Taken from Validate.js library](https://github.com/ansman/validate.js/blob/07f64ec81e8dd71d1553c20d750fbb3b7ed40018/validate.js#L1061)
```js
var emailCheck = /^[a-z0-9\u007F-\uffff!#$%&'*+\/=?^_`{|}~-]+(?:\.[a-z0-9\u007F-\uffff!#$%&'*+\/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z]{2,}$/i;

validate = function(email) {
	if (!email) return email;
	if (typeof email != "string") {
		console.error("(malicious?) non-string email:", email, new Error().stack);
		return "";
	}
	return emailCheck.test(email);
}

validate("mail@domain.com")
true
validate("tools@symbol.agency")
true
```

**Comment by adrienjoly** [comment]
Fixed by #100.

**Comment by znk** [comment]
<reserving the spot, still writing down my proposal>
I wanted to reach you back on that topic :laughing:
